rasterio/_base.pxd

```diff
@@ -1,9 +1,11 @@
 # Base class.
 
-cdef class DatasetReader:
-    # Read-only access to dataset metadata. No IO!
-    
-    cdef void *_hds
+include "gdal.pxi"
+
+
+cdef class DatasetBase:
+
+    cdef GDALDatasetH _hds
```

Comment:
@perrygeo @brendan-ward @geowurster a sneak peek at the kind of refactoring I want to do in #817.